mypy/server/subexpr.py

```diff
@@ -24,21 +24,50 @@ def __init__(self) -> None:
     def _visit_leaf(self, o: Expression) -> None:
         self.add(o)
```

Comment:
Can we remove `_visit_leaf` and just use `self.add(o)` directly in all methods you add below?